modules/mini-games/Race.lua

```diff
@@ -252,177 +279,177 @@ local function insideBox(box, pos)
+local respawn_car = Token.register(function(name)
     local player = variables["Dead_car"][name].player
-    local car =
-        surface[1].create_entity {
+    local position = surface[1].find_non_colliding_position('car', variables["Dead_car"][name].position, 5, 0.5)
+    position = position or variables["Dead_car"][name].position
+
+    local car = surface[1].create_entity {
         name = "car",
         direction = defines.direction.north,
-        position = variables["Dead_car"][name].position,
+        position = position,
         force = "player"
     }
+
+    car.operable = false
     car.set_driver(player)
     car.orientation = variables["Dead_car"][name].orientation
-    car.get_fuel_inventory().insert({name = variables["fuel"], count = 100})
+    car.get_fuel_inventory().insert{ name = variables["fuel"], count = 100 }
     cars[name] = car
 
     Permission_Groups.set_player_group(player, variables["Dead_car"][name].group)
-    invisabilty(car,name)
+    start_invincibility(car, name)
 
     variables["Dead_car"][name].car = car
-end
-
-token_for_car = Token.register(respawn_car)
-
+end)
 
+--- Triggered when an entity is destroyed, used to respawn the car
 local car_destroyed = function(event)
     local dead_car = event.entity
-    if dead_car.name == "car" then
-        local player = dead_car.get_driver().player
-        local name = player.name
-        if not variables["Dead_car"] then
-            variables["Dead_car"] = {}
-        end
-        if not variables["Dead_car"][name] then
-            variables["Dead_car"][name] = {}
-        end
-        variables["Dead_car"][name].player = player
-        variables["Dead_car"][name].position = dead_car.position
-        variables["Dead_car"][name].orientation = dead_car.orientation
-        variables["Dead_car"][name].group = Permission_Groups.get_group_from_player(player).name
-        player.character.destructible = false
-        Permission_Groups.set_player_group(player, "out_car")
-        task.set_timeout_in_ticks(180, token_for_car,name)
-        task.set_timeout_in_ticks(190, token_for_kill_biters,name)
-        task.set_timeout_in_ticks(480, token_for_stop_invins,name)
+    if dead_car.name ~= "car" then return end
+    if not variables["Dead_car"] then
+        variables["Dead_car"] = {}
+    end
+
+    -- Get the data for the car
+    local player = dead_car.get_driver().player
+    local name = player.name
+    local dead_car_data = variables["Dead_car"][name]
+    if not dead_car_data then
+        dead_car_data = {}
+        variables["Dead_car"][name] = dead_car_data
     end
-end
 
+    -- Update the data for the car
+    dead_car_data.player = player
+    dead_car_data.position = dead_car.position
+    dead_car_data.orientation = dead_car.orientation
+    dead_car_data.group = Permission_Groups.get_group_from_player(player).name
 
+    -- Make player invincible for a short time, then respawn the car
+    local offset = math.random(-30, 30)
```

Comment:
idk why their is a random espect involved?

Reply:
This is because the issue may be linked to the rotation of the cars, using find_non_coliding_possition does not account for this rotation. Allowing some randomness, upto half a seocnd, can allow one car to spawn before the other car preventing them from getting stuck because of rotation.

Reply:
yeah okey but it can as well as prevent it create it

Reply:
see discord